### PR TITLE
Move Octopus.Configuration interfaces and enum into tentacle and update references

### PR DESCRIPTION
**Pull request by scme0**
# Background

As part of #project-k8s-agent, I was adding the ability for Tentacle to save and source configuration from a configmap. During this work, I found that the `Octopus.Configuration` project has been merged back into `Octopus.Server` and the repo in github has been archived.

# Results

 This PR just removes the now defunct nuget package and moves the two interfaces and one enum into Tentacle. It's a relatively simple change, but it touches quite a few files to remove the old namespace.

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/main/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Octopus.Tentacle/Configuration/TentacleConfiguration.cs [hunk 1]
```diff
@@ -1,4 +1,3 @@
-#nullable enable
```
Comment (LukeButters) [inline source/Octopus.Tentacle/Configuration/TentacleConfiguration.cs:1]: Was this intentional?

**Review by LukeButters — APPROVED** [review]
LGTM,

Just double check if that removal of nullable was intentional, if so a small note on why would be worthwhile.

File: source/Octopus.Tentacle/Configuration/TentacleConfiguration.cs (hunk 1 above)
Comment (scme0) [reply source/Octopus.Tentacle/Configuration/TentacleConfiguration.cs:1]: Ah yeah I think that was rider deeming it unnecessary but I'll undo that change to keep this PR focused.

Comment (LukeButters) [reply source/Octopus.Tentacle/Configuration/TentacleConfiguration.cs]: Either way is fine :D

**Pull request merged**